bndtools.core/src/bndtools/central/Central.java

```diff
@@ -607,20 +601,31 @@ public static void refreshPlugins() throws Exception {
 	}
 
 	public static void refreshPlugin(Refreshable plugin) throws Exception {
-		if (plugin.refresh()) {
+		refreshPlugin(plugin, false);
+	}
+
+	public static void refreshPlugin(Refreshable plugin, boolean force) throws Exception {
+		if (plugin.refresh() || force) {
 			refreshFile(plugin.getRoot());
-			for (Project p : Central.getWorkspace()
-				.getAllProjects()) {
-				p.setChanged();
-				for (ModelListener l : getInstance().listeners)
-					l.modelChanged(p);
-			}
 			if (plugin instanceof RepositoryPlugin) {
 				repositoriesViewRefresher.repositoryRefreshed((RepositoryPlugin) plugin);
+			} else {
+				// If the Plugin was a RepositoryPlugin, the ViewRefresher will
+				// trigger the Project update. If not, we have to do it ourself
+				refreshProjects();
```

Comment:
What if multiple plugins refresh, then we call refreshProjects multiple times. If we are going to refresh the projects, we should do it once after all the plugins are refreshed not once for each plugin.

Reply:
The refreshPlugin method is only called by actions triggered in the UI. Thus they are sequential by nature. Who ever uses this method should be aware of what he is doing. 

The part where the Project refresh moved to the `RepositoriesViewRefresher` considers such a case and will only do it in the end, even if it is called multiple times (which might be a possibility).